### DISCUSSION TITLE
n2: failed to connect to amf on 10.53.1.2:38412. error="connection timed out" timeout=1361ms

### DISCUSSION THREAD

**Discussion by Guo-arch1017** [srsran/srsRAN_Project#832] [General Help] [answered]
operating system Ubuntu24.04
Usrp x310
I use docker compose -f docker/docker-compose.yml up
This is my profile
[docker.zip](https://github.com/user-attachments/files/17146332/docker.zip)

--== srsRAN gNB (commit ) ==--

n2: failed to connect to amf on 10.53.1.2:38412. error="connection timed out" timeout=1361ms

**Comment by pgawlowicz** [comment]
please follow this tutorial:
https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html

**Comment by NicolasGresset** [comment, marked as answer]
Make sure the configuration file used for the gNB specifies  the address that is effectively being used by the amf in your docker-compose. Make sure the gNB can ping the amf before even trying to start srsgnb. If not, you need to add a route between gNB desktop and CN desktop.
